lib/engine/game/base.rb

```diff
@@ -1932,17 +1932,20 @@ def after_par(corporation)
         place_home_token(corporation) if self.class::HOME_TOKEN_TIMING == :par
       end
 
+      # This is a hook to do something before company closes
+      def company_is_closing(company, silent = false)
```

Comment:
I can change the name of the method.   I pulled in the log message so that it can be modified, in the case of 18 India, the message includes the amount of money received for closing the company.

Reply:
Modified name to indicate after ability and reverted those cases I used this call for closing via event / phase.